Source/WebCore/Modules/fetch/FetchBodyOwner.cpp

```diff
@@ -138,47 +138,36 @@ void FetchBodyOwner::blob(Ref<DeferredPromise>&& promise)
         return;
     }
     m_isDisturbed = true;
-    m_body->blob(*this, WTFMove(promise), m_contentType);
+    m_body->blob(*this, WTFMove(promise));
 }
 
 void FetchBodyOwner::cloneBody(FetchBodyOwner& owner)
 {
     m_loadingError = owner.m_loadingError;
-
-    m_contentType = owner.m_contentType;
     if (owner.isBodyNull())
         return;
     m_body = owner.m_body->clone();
 }
 
 ExceptionOr<void> FetchBodyOwner::extractBody(FetchBody::Init&& value)
 {
-    auto result = FetchBody::extract(WTFMove(value), m_contentType);
+    String currentContentType = contentType();
+    auto result = FetchBody::extract(WTFMove(value), currentContentType);
     if (result.hasException())
         return result.releaseException();
+    m_headers->fastSet(HTTPHeaderName::ContentType, currentContentType);
```

Comment:
@youennf @annevk Question here 🙋 

Previously, we're passing `m_contentType` as a reference to `FetchBody::extract`'s second argument and mutate it directly when necessary. However, as I change the `contentType` function to _not_ return the value as a reference, it wouldn't update to the m_headers and I need to do the `m_headers->fastSet` here. Would you think it's better to return a pair of `(Body, NewContentType)` from the extract function instead? I wonder it would be helpful to remind the next person who will use the function that the content type would potentially be updated.

Reply:
Yeah, we could probably a later refactoring to have FetchBody::extract return all out parameters as return parameters. Probably not in this patch though.

Reply:
> I'm not sure I understand the question. Doesn't `FetchBody::extract` essentially return a pair? One of them via an output parameter? This looks okay to me.

oh, you are right 🙏  I didn't realize we can see it as an output parameter.

> Yeah, we could probably a later refactoring to have FetchBody::extract return all out parameters as return parameters. Probably not in this patch though.

yes, please let me know if you think we should refactor it and I will file an issue for it after this PR is merged 🙇